types/ramda/index.d.ts

```diff
@@ -402,6 +402,33 @@ export function clamp<T>(min: T): (max: T) => (value: T) => T;
 export function clone<T>(value: T): T;
 export function clone<T>(value: readonly T[]): T[];
 
+/**
+ * Splits a list into sub-lists,
+ * based on the result of calling a key-returning function on each element,
+ * and grouping the results according to values returned.
+ *
+ * See also {@link groupBy}, {@link partition}.
+ *
+ * @example
+ * ```typescript
+ * R.collectBy(R.prop('type'), [
+ *   {type: 'breakfast', item: '☕️'},
+ *   {type: 'lunch', item: '🌯'},
+ *   {type: 'dinner', item: '🍝'},
+ *   {type: 'breakfast', item: '🥐'},
+ *   {type: 'lunch', item: '🍕'}
+ * ]);
+ *
+ * // [ [ {type: 'breakfast', item: '☕️'},
+ * //     {type: 'breakfast', item: '🥐'} ],
+ * //   [ {type: 'lunch', item: '🌯'},
+ * //     {type: 'lunch', item: '🍕'} ],
+ * //   [ {type: 'dinner', item: '🍝'} ] ]
+ * ```
+ */
+export function collectBy<T, K extends PropertyKey>(fn: (value: T) => K, list: readonly T[]): T[][];
```

Comment:
It's better to give `fn` some more meaningful name if possible, other apis' `callback` should also give some more meaningful name instead of `fn`, `f`, `g`.